de.dlr.sc.virsat.model.extension.cefx.ui/src/de/dlr/sc/virsat/model/extension/cefx/ui/templates/DLRCEFXStudyCommandHelper.java

```diff
@@ -0,0 +1,277 @@
+		systemMassParameters.getMassAdapter().setDefaultValue(DEFAULT_MASS_ADAPTER);
+		systemMassParameters.getMassLaunchMax().setDefaultValue(DEFAULT_MASS_LAUNCH_MAX);
+		systemMassParameters.getMassPropellant().setDefaultValue(DEFAULT_MASS_PROPELLANT);
+		system.add(systemMassParameters);
+	}
+
+	public static void addProductTreeParameters(Concept conceptCEFX, ProductTree productTree) {
+
+	}
+	
+	public static void addProductTreeDomainParameters(Concept conceptCEFX, ProductTreeDomain productTreeDomain) {
+
+	}
+	
+	/**
+	 * @param conceptCEFX,  subSystem
+	 */
+	public static void addSubSystemParameters(Concept conceptCEFX, ElementConfiguration subSystem) {
+		SubSystemMassParameters subSystemMassParameters = new SubSystemMassParameters(conceptCEFX);
+		subSystemMassParameters.setName(MASS_PARAMETERS_NAME);
+		subSystem.add(subSystemMassParameters);
+	}
+
+	/**
+	 * @param conceptCEFX,  equipment
+	 */
+	public static void addEquipmentParameters(Concept conceptCEFX, ElementConfiguration equipment) {
+		EquipmentParameters equipmentParams = new EquipmentParameters(conceptCEFX);
+		equipmentParams.setMarginMaturity(DEFAULT_MARGIN_MATURITY);
+		equipment.add(equipmentParams);
+		
+		EquipmentMassParameters equipmentMassParameters = new EquipmentMassParameters(conceptCEFX);
+		equipmentMassParameters.setName(EQUIPMENT_MASS_PARAMETERS_NAME);
+		equipmentMassParameters.getMass().setDefaultValue(DEFAULT_EQUIPMENT_MASS_PARAMETERS);
+		equipment.add(equipmentMassParameters);
+
+	}
+	
+	/**
+	 * @param conceptCEFX,  elementDefinition
+	 */
+	public static void addElementDefinitionParameters(Concept conceptCEFX, ElementDefinition elementDefinition) {
+		EquipmentParameters equipmentParams = new EquipmentParameters(conceptCEFX);
+		equipmentParams.setMarginMaturity(DEFAULT_MARGIN_MATURITY);
+		elementDefinition.add(equipmentParams);
+		
+		EquipmentMassParameters equipmentMassParameters = new EquipmentMassParameters(conceptCEFX);
+		equipmentMassParameters.setName(EQUIPMENT_MASS_PARAMETERS_NAME);
+		equipmentMassParameters.getMass().setDefaultValue(DEFAULT_EQUIPMENT_MASS_PARAMETERS);
+		elementDefinition.add(equipmentMassParameters);	
+	}
+	
+	// Keep track of created disciplines
+	private static Set<String> createdDisciplines = new HashSet<>();
+	
+	/**
+	 * @param conceptCEFX,  disciplineName
+	 */
+	public static BeanDiscipline createDiscipline(VirSatTransactionalEditingDomain domain, String disciplineName) {
+		Discipline newDiscipline;
+		if (!createdDisciplines.contains(disciplineName)) {
+			RoleManagement roleManagement = domain.getResourceSet().getRoleManagement();
+			newDiscipline = RolesFactory.eINSTANCE.createDiscipline();
+			newDiscipline.setName(disciplineName);
+			newDiscipline.getUsers().add(UserRegistry.getInstance().getUserName());
+			Command addCommand = AddCommand.create(domain, roleManagement, RolesPackage.eINSTANCE.getRoleManagement_Disciplines(), newDiscipline);
+			domain.getCommandStack().execute(addCommand);
```

Comment:
This execute here is problematic as it already adds disciplines before the user actually decides to... You should rather add it to the compound command...

Reply:
Thanks for the comment, i fixed it in the upcomming commit. :)